### ISSUE TITLE
Multiple activities lifecycle issue

### ISSUE THREAD

**Issue by palmithor** [eneim/Toro#127]
Make sure to provide the following informations and check:

- [```im.ene.toro2:toro-extended:2.1.0``` ]

- [Any device I've tried it on ] 

- Logcat (if it was a fatal issue) - Not applicable

I’ve debugged a lot and I am having an issue with the lifecycle. 

In my case I have

MainActivity
	ViewPager
		Fragment w/ recycler view showing videos.

When I open another activity and go back, Toro.register(recyclerView); is called before onDestroy() of the closing activity. Therefore the Toro ActivityLifecycleCallback method onActivityDestroyed is called after the video has started playing and therefore does stop the video and put it in a state it cannot be restarted. 

When debugging the video actually starts if I stop on a breakpoint after Toro.register and before the onDestroy method. But when onDestroy for the second activity is called the video stops. If I scroll down (therefore recycle the view) and up again the video plays successfully.

I initialise Toro in the Application class, is there a way to disable the ActivityLifecycle callbacks for certain activities or what would be the right way to handle this?

I would really much appreciate the help, didn’t think this was an issue though - I am almost certain you know this :) If you need any help I would be glad to contribute.

**Comment by eneim** [comment]
Thanks for the issue. I have applied the mechanism to check the correct closing/openning Activity, but it seems to not be good enough. Let me take a look again.

**Comment by developerGM** [comment]
@eneim I have the same problem. When I close my activity and re-start it later, all videos are in pause mode. I need to close the app cleaning cache to reinitialize Toro.

**Comment by jpage4500** [comment]
@eneim - at quick glance it looks like someone fixed this issue (I'm having the same one too)

https://github.com/EligijusStarinskas/Toro/commit/6763e8f86984031012f34f9f0e0fa0eae6f59d43

Any chance you could pull this change into your library and publish? I can help test if you'd like

**Comment by jpage4500** [comment]
I submitted my own fix for this issue here: https://github.com/jpage4500/Toro/commit/654097d6bac1f3d7244089545c619651ac144af5

The key change is in onActivityDestroyed() to just stop/clear out only State's which belong to the Activity being destroyed.

I tested this change by adding a new sample Activity in the sample app (Tabs Sample) which illustrates the problem. 

To test, just open this Activity; click on any item to open another Activity; hit BACK to return to the prior Activity.. before, the prior video wouldn't resume and was in a bad state. Now, it resumes normally.

I'll try a pull request since I'd prefer to use the official library rather than my fork

**Comment by eneim** [comment]
@jpage4500 thanks a lot, currently I'm on the very last day of my year end vacation. I will back to Toro and take a look on you suggestion asap.

**Comment by eneim** [comment]
@jpage4500 Can you checkout my PR and see if it works? The idea is the same, but I make it work on View level (one Activity can have multiple RecyclerView in it).

**Comment by jpage4500** [comment]
Yep, I'll try this. I took a look and it seems like it'll work w/out having to add a new field to the SavedState class which is good.

**Comment by jpage4500** [comment]
@eneim - I'm trying to figure out how to use your new branch instead of the official maven repo version. I was using jitpack.io to include my forked version and that was working fine but this isn't working for your 'featurea/multi activity' branch.. any ideas how I can include it in my app?

Here's what I have now:

```
repositories {
...
    maven { url 'https://jitpack.io' }
}

    // compile 'im.ene.toro2:toro-extended:2.1.0' 
    compile 'com.github.eneim.Toro:toro-extended:featurea~multi-activity-SNAPSHOT' 
```

**Comment by eneim** [comment]
@jpage4500 I will find time to figure out a better way for feature-level update >.< it will be useful ...
